### PR TITLE
Link readme's installation section to doc/installation; improve installation doc

### PR DESCRIPTION
**Pull request by Boshen**
**Related issue (if exists):**
#2269 #2492 #2292

### CODE REVIEW

File: doc/installation.md [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
-To import the entire core set of functionality:
+## ES6
+
+Import all core functionality:
 
 ```js
 import Rx from 'rxjs/Rx';
 
-Rx.Observable.of(1,2,3)
+Rx.Observable.of(1,2,3).subscribe(console.log.bind(console));
```
Comment (Boshen) [inline doc/installation.md]: added subscribe and console.log so people can just copy and make it run

File: doc/installation.md [hunk 2]
```diff
@@ -65,43 +65,24 @@ Import operators and use them _manually_ you can do the following (this is also
...
 
 You can also use the above method to build your own Observable and export it from your own module.
 
 ### CommonJS with TypeScript
 If you recieve an error like `error TS2304: Cannot find name 'Promise'` or `error TS2304: Cannot find name 'Iterable'` when using RxJS you may need to install a supplemental set of typings.
 
-1. For [`typings`](https://github.com/typings/typings) users:
-
-    `typings install es6-shim --ambient`
+1. Using [`typings`](https://github.com/typings/typings): `typings install es6-shim --ambient`
```
Comment (Boshen) [inline doc/installation.md]: added `@types/es6-shim`

File: doc/installation.md [hunk 3]
```diff
@@ -65,43 +65,24 @@ Import operators and use them _manually_ you can do the following (this is also
...
+   or npm: `npm install --save-dev @types/es6-shim`
```
Comment (Boshen) [inline doc/installation.md]: added @types/es6-shim

**Comment by kwonoj** [comment]
This is related / possible dupe of long hanging https://github.com/ReactiveX/rxjs/pull/2292 . (Not sure why 2292 is stall until now though :/)

**Comment by staltz** [comment]
LGTM

**Pull request closed without merging**